eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Visitors/NameVisitor.cs

```diff
@@ -15,13 +17,31 @@ internal class NameVisitor : ScmLibraryVisitor
 {
     private const string ResourceTypeName = "ResourceType";
 
+    private readonly HashSet<CSharpType> _resourceUpdateModelTypes = new HashSet<CSharpType>();
+
     protected override ModelProvider? PreVisitModel(InputModelType model, ModelProvider? type)
     {
         if (type is not null && TryTransformUrlToUri(model.Name, out var newName))
         {
             type.Update(name: newName);
         }
 
+        if (type is not null && ManagementInputLibrary.IsResourceUpdateModel(model))
+        {
+            var enclosingResourceName = ManagementInputLibrary.FindEnclosingResourceNameForResourceUpdateModel(model);
```

Comment:
This is causing iteration of clients on every call.
Consider to populate the cache with Dictionary<InputModelType, resourceName>, also collect `HashSet<CSharpType> _resourceUpdateModelTypes` during the client traversal in InputLibrary.